src/main/java/nextstep/subway/favorite/application/FavoriteService.java

```diff
@@ -0,0 +1,58 @@
+package nextstep.subway.favorite.application;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityNotFoundException;
+import nextstep.subway.auth.domain.LoginMember;
+import nextstep.subway.favorite.domain.Favorite;
+import nextstep.subway.favorite.domain.FavoriteRepository;
+import nextstep.subway.favorite.dto.FavoriteRequest;
+import nextstep.subway.favorite.dto.FavoriteResponse;
+import nextstep.subway.member.application.MemberService;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class FavoriteService {
+
+    private final MemberService memberService;
+    private final StationService stationService;
+    private final FavoriteRepository favoriteRepository;
+
+    public FavoriteService(MemberService memberService, StationService stationService,
+                           FavoriteRepository favoriteRepository) {
+        this.memberService = memberService;
+        this.stationService = stationService;
+        this.favoriteRepository = favoriteRepository;
+    }
+
+    @Transactional
+    public FavoriteResponse createFavorite(LoginMember loginMember, FavoriteRequest favoriteRequest) {
+        Member member = memberService.findMember(loginMember.getId());
+        Station source = stationService.findStationById(favoriteRequest.getSource());
+        Station target = stationService.findStationById(favoriteRequest.getTarget());
+        Favorite favorite = favoriteRepository.save(new Favorite(member, source, target));
+
+        return new FavoriteResponse(favorite.getId(), favorite.getSource(), favorite.getTarget());
+    }
+
+    public List<FavoriteResponse> findAllFavorites(LoginMember loginMember) {
+        Member member = memberService.findMember(loginMember.getId());
+        List<Favorite> favorites = member.getFavorites();
+
+        return favorites.stream()
+                .map(favorite -> new FavoriteResponse(favorite.getId(), favorite.getSource(), favorite.getTarget()))
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void deleteFavorite(Long id) {
+        favoriteRepository.findById(id)
+                .orElseThrow(EntityNotFoundException::new);
+        favoriteRepository.deleteById(id);
```

Comment:
```suggestion
        favoriteRepository.delete(favorite);
```
위에서 조회한 entity 자체로 사용해도 될 것 같긴하네요 :)

아니면 의도적으로 이렇게 작성해주신걸까요 ㅎㅎ

Reply:
사실 이 부분에서 잠깐 고민을 했었는데요!

엔티티로 삭제하는것과 아이디값으로 삭제하는 부분 모두 동작은 동일했기에
가독성 좋은 방향을 선택해보자 판단했습니다.